### PR TITLE
Jonas add GitHub field and github command

### LINKED ISSUE #58: As a Student, keep track of my tutors/professors/friend's github username,
so that I find their github username easily.

### LINKED ISSUE #59: As a lazy user, I want to be able to open the github profile page of addresses in my address book with a command
so that I can view my friends/Teaching Assistants/Professors github projects easily.

### PR DESCRIPTION
**Pull request by Jonaspng**
Changes Done in this PR:
- Additional Optional field for person's github username
- github command which opens the github profile page of chosen person using default browser. (e.g command: github 1)

### CODE REVIEW

File: src/main/java/seedu/address/model/person/GithubUsername.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     * Github username cannot have multiple consecutive hyphens.
+     * Github username cannot begin or end with a hyphen.
+     * Maximum is 39 characters.
+     */
+    public static final String VALIDATION_REGEX = "^(?i)[a-z\\d](?:[a-z\\d]|-(?=[a-z\\d])){0,38}$";
+
+    public final String value;
+
+    /**
+     * Constructs a Location.
```
Comment (JJoeYi) [inline src/main/java/seedu/address/model/person/GithubUsername.java]: Missed a change here. Could it be "Constructs a GithubUsername"?

**Review by JJoeYi — COMMENTED** [review]
LGTM. Just a nit.

File: src/main/java/seedu/address/model/person/GithubUsername.java (hunk 1 above)
Comment (Jonaspng) [reply src/main/java/seedu/address/model/person/GithubUsername.java]: Opps. Thanks for pointing out.

**Review by nitant-p — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/GithubCommand.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import seedu.address.model.person.Person;
+
+/**
+ * Opens Github User profile page on browser.
+ */
+public class GithubCommand extends Command {
+
+    public static final String COMMAND_WORD = "github";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Opens up Github Profile page associated to chosen "
```
Comment (GarethOng) [inline src/main/java/seedu/address/logic/commands/GithubCommand.java]: I believe you can be more specific by specifying who the Github Profile page associated to.
```suggestion
    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Opens up Github Profile page associated to index specified "
```

**Review by GarethOng — COMMENTED** [review]
LGTM!

**Review by normkoh — COMMENTED** [review]
LGTM!

**Pull request merged**